src/Vizkit3dWorld.cpp

```diff
@@ -36,9 +39,8 @@ Vizkit3dWorld::Vizkit3dWorld(std::string path,
 {
     loadGazeboModelPaths(modelPaths);
 
-    int argc = 1;
-    char const*argv[] = { "vizkit3d_world" };
-    app = new QApplication(argc, const_cast<char**>(argv));
+    if (!qApp)
+        app = new QApplication(argc, const_cast<char**>(argv));
```

Comment:
You don't need to keep a reference to this pointer, qt does that for you

Reply:
We are not using the app anymore. I guess @romulogcerqueira can use something like the code below: Is it ok for you @g-arjones?
```c++
 if (!qApp) new QApplication(argc, argv);
```

Reply:
That or just not using a pointer at all.. Whatever you think is best

Reply:
If you do decide to use a pointer, don't forget to delete it

Reply:
Ok.

Reply:
> This could (should?) even be allocated in the stack
As I can remember, I tried to allocate in the stack, but I could not. Usually, the QApplication is instantiated in the stack in the main function. But I don't know how to proceed into a library.

Reply:
Do you remember why you couldn't allocate it in the stack? I can't see a reason

Reply:
I don't remember why, but if I try to allocate the QApplication before the function **loadGazeboModelPaths**, also the program crashes. As far I can remeber, I can't compile the program as well. Maybe I did something wrong!

Reply:
Well, if you could not compile it then it was definitely a syntax error..

Reply:
I did a test allocating the QApplication before `loadGazeboModelPaths` and it worked.

Reply:
Nice! @romulogcerqueira

Reply:
Why would QApplication depend on loadGazeboModelPaths ?

Reply:
It does not depend. Maybe, I think the problem was the scope of argc/argv. My mistake.